### ISSUE TITLE
Consistency in how we refer and link to the main STAC constructs

### ISSUE THREAD

**Issue by cholmes** [radiantearth/stac-spec#973]
Throughout the spec text, a STAC Item may be referred to as an item, Item, `item`, 'Item'. And it may not have a link, or it may link to item-spec/README.md or item-spec/item-spec.md  The same is true with collection and catalog, and also things like Assets and Links (capitalize? prefix with STAC?). And also fields, sometimes they get the `code` treatment, sometimes not, sometimes a quote. 

We should decide on conventions of when to link, and how to consistently refer to things.

This technically could be done between RC1 and final, as it does not affect the actual spec at all, but scheduling it for before.

**Comment by cholmes** [comment]
Item, optionally STAC Item. 

Link first occurance in spec, possibly in first H2 (if it's really long), don't keep repeating, unless to a specific section. To spec itself.

**Comment by cholmes** [comment]
One thing I've encountered here is that we often use 'catalog' to refer to an overall STAC implementation. I started trying to replace those with 'STAC implementation', but it didn't really work, with 'static catalog' and 'self-contained catalog', so mostly went back to Catalog if it's the json file, but catalog if it's a full stac implementation.

But then Collection was somewhat similar, though for some reason there I leaned more to calling a set of items in Collection a 'Collection' instead of a 'collection', which is less consistent. But it felt like a Collection naturally includes all the Items in it, while a Catalog is just a structural element. It'd be good to get decision / guidance on how to tackle those. We maybe should also clarify somewhere.
